### PR TITLE
54 add update user data functionality

### LINKED ISSUE #54: Add Update user data functionality
**Short description:**
The goal of this issue is to create the possibility to update the user personal information.

**Describe the solution you'd like**

- Creation of a token protected custom road inside the Rest API with all the new data transferred inside the body.
- No need to confirm operation by e-mail
- Update the actual settings screen by adding user personal data visualization
- Add a new Flutter button to call the new update API road.

### PR DESCRIPTION
**Pull request by LucasTesnier**
# Purpose

The purpose of this pull request is to add the update user data functionality.

Fixes #54 

## Added features

* The update user data functionality have been added !
 * You can customize your Username, Password and Email !
 * When a new e-mail have been set, a new verification mail is send.
 * The road is protected by token authentication
 
* An user data customization visualization have been implemented in Flutter side ! 
* A new Flutter button is now accessible to publish updated information :)

### CODE REVIEW

File: server/server_app.js [hunk 1]
```diff
@@ -292,6 +293,58 @@ app.post('/api/user/resetPassword', async (req, res, next) => {
   return res.json('Verification e-mail sended.')
 })
 
+/**
+ * Post request to update user personal data.
+ * body.username -> User name
+ * body.email -> User mail
+ * body.password -> User password
+ * Road protected by token authentification
+ * An new e-mail verification is send when e-mail is updated.
```
Comment (GuyomT) [inline server/server_app.js]: sent :)

**Review by GuyomT — DISMISSED** [review]
LGTM

**Review by TomDUVAL-MAHE — DISMISSED** [review]

File: server/server_app.js (hunk 1 above)
Comment (LucasTesnier) [reply server/server_app.js]: fixed

File: server/server_app.js [hunk 2]
```diff
@@ -292,6 +293,58 @@ app.post('/api/user/resetPassword', async (req, res, next) => {
   return res.json('Verification e-mail sended.')
```
Comment (GuyomT) [inline server/server_app.js]: NOOOOOOOOOOOOOOO
Verbe irrégulier : send - sent - sent

**Review by GuyomT — DISMISSED** [review]

File: server/server_app.js (hunk 2 above)
Comment (LucasTesnier) [reply server/server_app.js]: fuuuuckk)cZEMLK NZSVmln jkez

**Review by GuyomT — APPROVED** [review]
LGTMMMMMMMMMMMMMMMMMMMMMMMMMM !!

**Review by tbellicha — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**